### DISCUSSION TITLE
docker compose up not working

### DISCUSSION THREAD

**Discussion by cb-sudheer** [juspay/hyperswitch#2030] [Q&A] [answered]
I am trying to set the hyperswitch in locally using the docker-compose up, but hyper switch server is failing with the below error

   Compiling router v0.2.0 (/app/crates/router)

error: could not compile `router` (lib)

Caused by:

  process didn't exit successfully: `rustc --crate-name router --edition=2021 crates/router/src/lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --crate-type lib --emit=dep-info,metadata,link -C embed-bitcode=no -C debuginfo=2 --cfg 'feature="accounts_cache"' --cfg 'feature="default"' --cfg 'feature="dummy_connector"' --cfg 'feature="kv_store"' --cfg 'feature="olap"' --cfg 'feature="oltp"' --cfg 'feature="payouts"' --cfg 'feature="stripe"' -C metadata=5bc49e3b2b6ddd0b -C extra-filename=-5bc49e3b2b6ddd0b --out-dir /cargo_build_cache/debug/deps -C incremental=/cargo_build_cache/debug/incremental -L dependency=/cargo_build_cache/debug/deps --extern actix=/cargo_build_cache/debug/deps/libactix-3f167f5f46e88488.rmeta --extern actix_cors=/cargo_build_cache/debug/deps/libactix_cors-929ea6ba8cf8ca1e.rmeta --extern actix_multipart=/cargo_build_cache/debug/deps/libactix_multipart-cfe158db1cc93e0b.rmeta --extern actix_rt=/cargo_build_cache/debug/deps/libactix_rt-eb5b10840abc4f31.rmeta --extern actix_web=/cargo_build_cache/debug/deps/libactix_web-302d40e234c0af87.rmeta --extern api_models=/cargo_build_cache/debug/deps/libapi_models-ebebbdc03a5c7c1e.rmeta --extern async_bb8_diesel=/cargo_build_cache/debug/deps/libasync_bb8_diesel-c957986e3286fead.rmeta --extern async_trait=/cargo_build_cache/debug/deps/libasync_trait-fdd312eb5fd1514e.so --extern base64=/cargo_build_cache/debug/deps/libbase64-64ffb08cb1d39076.rmeta --extern bb8=/cargo_build_cache/debug/deps/libbb8-1c4870690d5b98a2.rmeta --extern blake3=/cargo_build_cache/debug/deps/libblake3-a8b97bb37a8ad041.rmeta --extern bytes=/cargo_build_cache/debug/deps/libbytes-a354a96e7604a4ce.rmeta --extern cards=/cargo_build_cache/debug/deps/libcards-68785a2027443166.rmeta --extern clap=/cargo_build_cache/debug/deps/libclap-4629476104ab487e.rmeta --extern common_utils=/cargo_build_cache/debug/deps/libcommon_utils-57e70deec37e1173.rmeta --extern config=/cargo_build_cache/debug/deps/libconfig-b310878830859370.rmeta --extern diesel=/cargo_build_cache/debug/deps/libdiesel-7223246957d322b2.rmeta --extern diesel_models=/cargo_build_cache/debug/deps/libdiesel_models-0d321f34829e7544.rmeta --extern dyn_clone=/cargo_build_cache/debug/deps/libdyn_clone-81ae38d805213545.rmeta --extern encoding_rs=/cargo_build_cache/debug/deps/libencoding_rs-1a3c46c8cf690459.rmeta --extern error_stack=/cargo_build_cache/debug/deps/liberror_stack-597006844eb18162.rmeta --extern external_services=/cargo_build_cache/debug/deps/libexternal_services-b0bde01397ddbab3.rmeta --extern futures=/cargo_build_cache/debug/deps/libfutures-7ecfc854fdd22281.rmeta --extern hex=/cargo_build_cache/debug/deps/libhex-359206462e010d72.rmeta --extern http=/cargo_build_cache/debug/deps/libhttp-b779fb5128a0f2e5.rmeta --extern image=/cargo_build_cache/debug/deps/libimage-0471bded85b76b1e.rmeta --extern infer=/cargo_build_cache/debug/deps/libinfer-116c31193b8ec15f.rmeta --extern josekit=/cargo_build_cache/debug/deps/libjosekit-1056d42a16738177.rmeta --extern jsonwebtoken=/cargo_build_cache/debug/deps/libjsonwebtoken-760783fed4254cdc.rmeta --extern literally=/cargo_build_cache/debug/deps/libliterally-43685da262030342.rmeta --extern masking=/cargo_build_cache/debug/deps/libmasking-f79f57c8a4c0ecf9.rmeta --extern maud=/cargo_build_cache/debug/deps/libmaud-b17b673ac54e5c98.rmeta --extern mime=/cargo_build_cache/debug/deps/libmime-1b34e86192063b1b.rmeta --extern nanoid=/cargo_build_cache/debug/deps/libnanoid-0121dda665b66f25.rmeta --extern num_cpus=/cargo_build_cache/debug/deps/libnum_cpus-8bb435845646f547.rmeta --extern once_cell=/cargo_build_cache/debug/deps/libonce_cell-66d12da63e46fdd9.rmeta --extern qrcode=/cargo_build_cache/debug/deps/libqrcode-1eac1e03bf6dbb5d.rmeta --extern rand=/cargo_build_cache/debug/deps/librand-9a70314acc43a230.rmeta --extern redis_interface=/cargo_build_cache/debug/deps/libredis_interface-2fe582a4eee73cde.rmeta --extern regex=/cargo_build_cache/debug/deps/libregex-369a3bf442b65879.rmeta --extern reqwest=/cargo_build_cache/debug/deps/libreqwest-d4ce48e79628b867.rmeta --extern ring=/cargo_build_cache/debug/deps/libring-af30df7a94a630e3.rmeta --extern router_derive=/cargo_build_cache/debug/deps/librouter_derive-dde12f945ef5fb40.so --extern router_env=/cargo_build_cache/debug/deps/librouter_env-3b0080c85dc33194.rmeta --extern roxmltree=/cargo_build_cache/debug/deps/libroxmltree-c7de83cfa596f2d4.rmeta --extern serde=/cargo_build_cache/debug/deps/libserde-546f10df7a421bbe.rmeta --extern serde_json=/cargo_build_cache/debug/deps/libserde_json-9723f71c08e8e3e7.rmeta --extern serde_path_to_error=/cargo_build_cache/debug/deps/libserde_path_to_error-634d008553b31b91.rmeta --extern serde_qs=/cargo_build_cache/debug/deps/libserde_qs-3fc987ab4a1db857.rmeta --extern serde_urlencoded=/cargo_build_cache/debug/deps/libserde_urlencoded-91a4fb4646979e3c.rmeta --extern serde_with=/cargo_build_cache/debug/deps/libserde_with-12d823d5dd8d71f3.rmeta --extern signal_hook=/cargo_build_cache/debug/deps/libsignal_hook-25cdfae694e2abd6.rmeta --extern signal_hook_tokio=/cargo_build_cache/debug/deps/libsignal_hook_tokio-99f364a625f97855.rmeta --extern storage_impl=/cargo_build_cache/debug/deps/libstorage_impl-c79df7c017bc9f3c.rmeta --extern strum=/cargo_build_cache/debug/deps/libstrum-ad830c8bfede4f58.rmeta --extern thiserror=/cargo_build_cache/debug/deps/libthiserror-7e09a928822ea9cb.rmeta --extern time=/cargo_build_cache/debug/deps/libtime-f6d6481293446aee.rmeta --extern tokio=/cargo_build_cache/debug/deps/libtokio-a158fea4d66808b8.rmeta --extern url=/cargo_build_cache/debug/deps/liburl-cdbb065e6cc850bf.rmeta --extern utoipa=/cargo_build_cache/debug/deps/libutoipa-53ac387b889ce0e5.rmeta --extern utoipa_swagger_ui=/cargo_build_cache/debug/deps/libutoipa_swagger_ui-e23c28e3f2be5098.rmeta --extern uuid=/cargo_build_cache/debug/deps/libuuid-f00c3f09f44f3b46.rmeta -Funsafe_code '-Wclippy::as_conversions' '-Wclippy::expect_used' '-Wclippy::missing_panics_doc' '-Wclippy::panic_in_result_fn' '-Wclippy::panic' '-Wclippy::panicking_unwrap' '-Wclippy::todo' '-Wclippy::unimplemented' '-Wclippy::unreachable' '-Wclippy::unwrap_in_result' '-Wclippy::unwrap_used' '-Wclippy::use_self' -Wrust_2018_idioms -Wunused_qualifications -L native=/cargo_build_cache/debug/build/zstd-sys-8cd164fb2b8ce80e/out -L native=/usr/lib/aarch64-linux-gnu -L native=/cargo_build_cache/debug/build/ring-3498e904d81c343f/out -L native=/cargo_build_cache/debug/build/blake3-f60bae3d42249394/out` (signal: 9, SIGKILL: kill)

   Compiling router v0.2.0 (/app/crates/router)

error: could not compile `router` (lib)

**Comment by cb-harishbharadwaj** [comment, marked as answer]
Please upgrade to 4.21.1 Docker Desktop. @cb-sudheer 
Also only one error came which is 
`Unable to deserialize application configuration: multiple_api_version_supported_connectors.supported_connectors: invalid type: sequence, expected a string for key `multiple_api_version_supported_connectors.supported_connectors` in config/docker_compose.toml`

**Reply by cb-harishbharadwaj** [reply]
The above error can be fixed temporarily by changing the below 
-supported_connectors = ["braintree"]
+supported_connectors = "braintree"

**Reply by SanchithHegde** [reply]
Thanks for helping address the issue @cb-harishbharadwaj!

If it isn't asking for too much, could you please open a PR fixing these config files?

https://github.com/juspay/hyperswitch/blob/c2aa0142ed5af0b5fcf21b35cb129addd92c6125/config/docker_compose.toml#L291
https://github.com/juspay/hyperswitch/blob/c2aa0142ed5af0b5fcf21b35cb129addd92c6125/config/config.example.toml#L403
https://github.com/juspay/hyperswitch/blob/c2aa0142ed5af0b5fcf21b35cb129addd92c6125/loadtest/config/development.toml#L210

**Comment by SanchithHegde** [comment]
Closing this as resolved.

**Reply by cb-sudheer** [reply]
@SanchithHegde Is there a way to manage the memory in mac os?

**Reply by SanchithHegde** [reply]
For Docker? If you're using the Docker Desktop application, you can go to Settings > Resources and configure resource usage thresholds there.

No idea if this can be set with the help of a config file though.
